src/ray/core_worker/actor_manager.cc

```diff
@@ -178,41 +191,11 @@ bool ActorManager::AddActorHandle(std::unique_ptr<ActorHandle> actor_handle,
     return inserted;
   }
 
-  if (inserted) {
-    // Register a callback to handle actor notifications.
-    auto actor_notification_callback =
-        std::bind(&ActorManager::HandleActorStateNotification,
-                  this,
-                  std::placeholders::_1,
-                  std::placeholders::_2);
-    RAY_CHECK_OK(gcs_client_->Actors().AsyncSubscribe(
-        actor_id,
-        actor_notification_callback,
-        [this, actor_id, cached_actor_name](Status status) {
-          if (status.ok() && !cached_actor_name.empty()) {
-            {
-              absl::MutexLock lock(&cache_mutex_);
-              cached_actor_name_to_ids_.emplace(cached_actor_name, actor_id);
-            }
-          }
-        }));
-  }
-
   return inserted;
 }
 
 void ActorManager::OnActorKilled(const ActorID &actor_id) {
```

Comment:
> 

I prefer to keep this method, otherwise I need to expose the `MarkActorKilledOrOutOfScope` interface, which depends on `std::shared_ptr<ActorHandle> actor_handle`, and I think exposing `actor_id` is better than `actor_handle`.

Reply:
Why not just making `MarkActorKilledOrOutOfScope ` accept actor id?

Reply:
It will cause dead lock if it accept actor_id，because the both GetActorHandle(invoked inside MarkActorKilledOrOutOfScope(actor_id)) and WaitForActorOutOfScope are going to get the mutex lock.